### ISSUE TITLE
Allow sssd to retrieve sudo rules of local users whose sudo rules stored in ldap server

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2498]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2494
- Created at 2014-11-15 02:07:29 by [dpal](https://pagure.io/user/dpal)
- Closed at 2019-06-13 23:26:06 as worksforme
- Assigned to nobody
---

Ticket was cloned from Red Hat Bugzilla (product *RHEL RFE*): [Bug 1132264](https://bugzilla.redhat.com/show_bug.cgi?id=1132264)

*Please note that this Bug is private and may not be accessible as it contains confidential Red Hat customer information.*

    Description of problem:
    - In RHEL 6.5 configured as ldap-client using sssd, there are some local users
    whose sudo rules are stored/defined on LDAP server. sssd is not fetching those
    sudo rules from ldap server if the users are local to system.
    - At this stage, SSSD only supports sudo rules and users being in the same
    domain. Add this feature of retrieving sudo rules for local users in sssd.
    - The local users do not have account ( duplicate account ) on ldap server.
    
    Version-Release number of selected component (if applicable):
    sssd-1.9.2-129.el6_5.4.x86_64
    
    How reproducible:
    
    
    Steps to Reproduce:
    1. configure rhel6 as ldap client, and also configure it to fetch sudo rules
    from ldap server
    2. create local users on rhel6 and add define their sudo rules on ldap server.
    3. Try to fetch these sudo rules from rhel6 client.
    
    Actual results:
    
    local user is not able to run the sudo rules:
    ]# sudo -l -U testuser
    User testuser is not allowed to run sudo on localsystem
    
    Expected results:
    Local user should be able to run the sudo rules.
    
    Additional info:
    - At this stage, this feature is available with nss-pam-ldap. As per discussion
    with Jakub, this feature will be included in version 1.13.

### Comments

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2014-11-21 17:25:37](https://pagure.io/SSSD/sssd/issue/2494#238067)**

Depends on the feature to manage local users.

blockedby:  => 
blocking:  => 
changelog:  => 
coverity:  => 
design:  => 
design_review:  => 0
feature_milestone:  => 
fedora_test_page:  => 
mark: no => 0
milestone: NEEDS_TRIAGE => SSSD 1.13 beta
review: True => 0
selected:  => 
testsupdated:  => 0

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2014-11-21 17:26:00](https://pagure.io/SSSD/sssd/issue/2494#238068)**

Fields changed

blockedby:  => #2228

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2015-02-10 14:41:33](https://pagure.io/SSSD/sssd/issue/2494#238069)**

Fields changed

milestone: SSSD 1.13 beta => SSSD 1.13 backlog

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2015-02-12 20:13:48](https://pagure.io/SSSD/sssd/issue/2494#238070)**

Mass-moving tickets not planned for the 1.13 release to 1.14

milestone: SSSD 1.13 backlog => SSSD 1.14 beta

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2015-07-28 20:06:23](https://pagure.io/SSSD/sssd/issue/2494#238071)**

Fields changed

priority: major => minor
sensitive:  => 0

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2016-02-16 14:20:31](https://pagure.io/SSSD/sssd/issue/2494#238072)**

This will happen for free when we support local users.

milestone: SSSD 1.14 beta => SSSD 1.15 beta

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2017-02-24 14:57:30](https://pagure.io/SSSD/sssd/issue/2494#238073)**

**Metadata Update from @dpal**:
- Issue marked as depending on: #2228
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2017-08-09 17:48:02](https://pagure.io/SSSD/sssd/issue/2494#456236)**

Putting into 1.16 (next major upstream). I suspect this is actually done since the files provider was implemented, but we need to re-test.

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2017-08-09 17:48:02](https://pagure.io/SSSD/sssd/issue/2494#456237)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review reset (from 0)
- Custom field sensitive reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None
- Issue set to the milestone: SSSD 1.16.0 (was: SSSD Future releases (no date set yet))

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2017-10-19 20:42:31](https://pagure.io/SSSD/sssd/issue/2494#473652)**

Since we are required to release a new upstream tarball no later than Friday Oct-20, I'm moving tickets that will not be closed by that date to the next milestone, 1.16.1

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2017-10-19 20:42:35](https://pagure.io/SSSD/sssd/issue/2494#473653)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue set to the milestone: SSSD 1.16.1 (was: SSSD 1.16.0)

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2017-12-18 20:33:42](https://pagure.io/SSSD/sssd/issue/2494#485199)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue tagged with: postpone-to-2-0

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2018-01-08 18:23:15](https://pagure.io/SSSD/sssd/issue/2494#487250)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue **un**tagged with: postpone-to-2-0
- Issue set to the milestone: SSSD 2.0 (was: SSSD 1.16.1)

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2018-08-13 11:11:45](https://pagure.io/SSSD/sssd/issue/2494#525886)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue set to the milestone: SSSD 2.1 (was: SSSD 2.0)

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2019-02-22 15:53:26](https://pagure.io/SSSD/sssd/issue/2494#555734)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue set to the milestone: SSSD 2.2 (was: SSSD 2.1)

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2019-06-13 23:26:06](https://pagure.io/SSSD/sssd/issue/2494#576132)**

This has been working for a long time..

---

**Comment from [dpal](https://pagure.io/user/dpal) at [2019-06-13 23:26:06](https://pagure.io/SSSD/sssd/issue/2494#576133)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue close_status updated to: worksforme
- Issue status updated to: Closed (was: Open)
